### PR TITLE
add an option to exit sim when the first workload finishes

### PR DESCRIPTION
**Pull request by SleepBook**
A little bit unsure about whether this is done in a correct way. With this patch the run file accept something like `--exit_cond <first_done|last_done>` but won't accept `--exit_cond <0|1>`. Even though the default for this config is defined as `0` (last_done) in the `general.param.def` file.

### CODE REVIEW

File: src/param_parser.c [hunk 1]
```diff
@@ -79,6 +79,7 @@ the program.  This way, an exact duplicate run can be performed.
 const char* help_options[]   = {"-help", "-h", "--help",
                               "--h"}; /* cmd-line help options strings */
 const char* sim_mode_names[] = {"uop", "full"};
+const char* exit_cond[]      = {"last_done", "first_done"};
```
Comment (siavashzk) [inline src/param_parser.c:82]: rename to "exit_cond_names" for consistency

File: src/sim.c [hunk 2]
```diff
@@ -613,7 +613,8 @@ void uop_sim() {
 
 void full_sim() {
   uns8 proc_id;
-  Flag all_sim_done = FALSE;
+  Flag all_sim_done         = FALSE;
+  Flag sim_process_finished = FALSE;
```
Comment (siavashzk) [inline src/sim.c:617]: rename to "any_sim_done"

File: src/sim.c [hunk 3]
```diff
@@ -651,7 +652,14 @@ void full_sim() {
   clear_stats = trigger_create("CLEAR_STATS", CLEAR_STATS, TRIGGER_ONCE);
 
   /* main loop */
-  while(!trigger_fired(sim_limit) && !all_sim_done) {
+  while(!trigger_fired(sim_limit)) {
+    // sim control
+    if(all_sim_done)
+      break;
+    else if(EXIT_COND == FIRST_DONE) {
```
Comment (siavashzk) [inline src/sim.c:659]: if ((EXIT_COND == LAST_DONE && all_sim_done)
    || EXIT_COND == FIRST_DONE && any_sim_done))
  break;

**Review by siavashzk — APPROVED** [review]

**Comment by SleepBook** [comment]
rebased to the upstream HEAD

**Pull request merged**